src/xpi.js

```diff
@@ -134,28 +134,27 @@ export default class Xpi {
   }
 
   getFilesByExt(...extensions) {
-    return new Promise((resolve, reject) => {
 
-      for (let ext of extensions) {
-        if (ext.indexOf('.') !== 0) {
-          throw new Error("File extension must start with '.'");
-        }
+    for (let ext of extensions) {
+      if (ext.indexOf('.') !== 0) {
+        // Use Promise.reject as outside a then a throw
+        // wouldn't be caught.
+        return Promise.reject(new Error("File extension must start with '.'"));
```

Comment:
As per the comment. We're not in the Promise constructor callback or a then so this is not part of the Promise chain at this point. I can improve the comment.